code/game/objects/items/incense.dm

```diff
@@ -176,6 +223,16 @@
 		var/mob/M = loc
 		M.update_inv_hands()
 
+/obj/item/incense_stick/proc/set_fragrance(var/newfrag)
+	if(fragrance == newfrag)
+		return FALSE
+	fragrance = newfrag
+	for(var/path in typesof(/obj/item/incense_stick))
+		var/obj/item/incense_stick/IS = new path
+		if(fragrance == IS.fragrance)
```

Comment:
It would have to be a nested associative list because fragrance is also used to look up oilbox types lower down, we had this conversation in the Discord